cli/installer/install-azd.ps1

```diff
@@ -76,33 +113,68 @@ if ($DryRun) {
     exit 0
 }
 
-New-Item -ItemType Directory -Path $InstallFolder -Force | Out-Null
+if (!(Test-Path $InstallFolder)) {
+    New-Item -ItemType Directory -Path $InstallFolder -Force | Out-Null
+}
 
 $tempFolder = "$([System.IO.Path]::GetTempPath())$([System.IO.Path]::GetRandomFileName())"
+Write-Verbose "Creating temporary folder for downloading and extracting binary: $tempFolder"
 New-Item -ItemType Directory -Path $tempFolder | Out-Null
 
 Write-Verbose "Downloading latest build from $downloadUrl" -Verbose:$Verbose
-$zipFileName = "$tempFolder\azd-windows-amd64.zip"
+$releaseArchiveFileName = "$tempFolder/$packageFilename"
 try {
-    Invoke-WebRequest -Uri $downloadUrl -OutFile $zipFileName -TimeoutSec $DownloadTimeoutSeconds
+    Invoke-WebRequest -Uri $downloadUrl -OutFile $releaseArchiveFileName -TimeoutSec $DownloadTimeoutSeconds
 } catch {
     Write-Error "Error downloading $downloadUrl"
     Write-Error $_
     exit 1
 }
 
-Write-Verbose "Unzipping artifacts" -Verbose:$Verbose
-try {
-    Expand-Archive -Path $zipFileName -DestinationPath $tempFolder/unzip
-} catch {
-    Write-Error "Cannot expand $zipFileName"
-    Write-Error $_
-    exit 1
+Write-Verbose "Decompressing artifacts" -Verbose:$Verbose
+if ($extension -eq 'zip') {
+    try {
+        Expand-Archive -Path $releaseArchiveFileName -DestinationPath $tempFolder/decompress
+    } catch {
+        Write-Error "Cannot expand $releaseArchiveFileName"
+        Write-Error $_
+        exit 1
+    }
+} elseif ($extension -eq 'tar.gz') {
+    Write-Verbose "Extracting to $tempFolder/decompress"
+    New-Item -ItemType Directory -Path "$tempFolder/decompress" | Out-Null
+    Write-Host "tar -zxvf $releaseArchiveFileName -C `"$tempFolder/decompress`" $binFilename"
+    tar -zxvf $releaseArchiveFileName -C "$tempFolder/decompress" $binFilename
+
+    if ($LASTEXITCODE) {
+        Write-Error "Cannot expand $releaseArchiveFileName"
+        exit $LASTEXITCODE
+    }
 }
 
 Write-Verbose "Installing azd in $InstallFolder" -Verbose:$Verbose
+
+$outputFilename = "$InstallFolder/azd.exe"
+if (isLinuxOrMac) {
+    $outputFilename = "$InstallFolder/azd"
+}
+
 try {
-    Copy-Item "$tempFolder/unzip/azd-windows-amd64.exe" "$InstallFolder/azd.exe" | Out-Null
+    if (isLinuxOrMac) {
+        test -w "$InstallFolder/"
+        if ($LASTEXITCODE) {
+            Write-Host "Writing to $InstallFolder/ requires elevated permission. You may be prompted to enter credentials."
+            sudo cp "$tempFolder/decompress/$binFilename" $outputFilename
```

Comment:
I assume cp here isn't an alias but instead the native cp command.

Reply:
It is the native `cp` command (`/usr/bin/cp` in Ubuntu 20.04). If `sudo` and/or `cp` are overridden in pwsh then behavior here will not be predictable.